pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -676,6 +668,38 @@ async def _map_messages(
 
         return system_instruction, contents
 
+    async def _map_tool_return(self, part: ToolReturnPart) -> list[PartDict]:
+        """Map a ToolReturnPart to Google API format, handling multimodal content."""
+        response_parts = part.model_response_parts()
+        has_multimodal = any(isinstance(p, (FileUrl, BinaryContent)) for p in response_parts)
+        if has_multimodal:
+            text_parts = [p for p in response_parts if isinstance(p, str)]
+            response_text = ' '.join(text_parts) if text_parts else ''
+            result: list[PartDict] = [
+                {
+                    'function_response': {
+                        'name': part.tool_name,
+                        'response': {'result': response_text} if response_text else {},
+                        'id': part.tool_call_id,
+                    }
+                }
+            ]
+            for item in response_parts:
+                if isinstance(item, (FileUrl, BinaryContent)):
+                    user_prompt = UserPromptPart(content=[item])  # pyright: ignore[reportArgumentType]
+                    result.extend(await self._map_user_prompt(user_prompt))
```

Comment:
Why don't we use the `FunctionResponseDict.parts` field instead of sending them like we used to?